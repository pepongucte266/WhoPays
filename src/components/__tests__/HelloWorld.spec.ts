import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import HelloWorld from '../HelloWorld.vue'

describe('HelloWorld', () => {
  it('renders msg prop', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'Hello Vitest' } })
    expect(wrapper.text()).toContain('Hello Vitest')
  })

  it('h1 có class green', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'Test' } })
    const h1 = wrapper.find('h1')
    expect(h1.classes()).toContain('green')
  })

  it('có 2 link đúng href', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'Test' } })
    const links = wrapper.findAll('a')
    expect(links.length).toBe(2)
    expect(links[0].attributes('href')).toContain('vite')
    expect(links[1].attributes('href')).toContain('vuejs')
  })

  it('matches snapshot', () => {
    const wrapper = mount(HelloWorld, { props: { msg: 'Snapshot' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
