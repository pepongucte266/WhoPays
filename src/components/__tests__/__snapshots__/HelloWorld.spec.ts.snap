// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > matches snapshot 1`] = `
"<div data-v-e17ea971="" class="greetings">
  <h1 data-v-e17ea971="" class="green">Snapshot</h1>
  <h3 data-v-e17ea971=""> Youâ€™ve successfully created a project with <a data-v-e17ea971="" href="https://vite.dev/" target="_blank" rel="noopener">Vite</a> + <a data-v-e17ea971="" href="https://vuejs.org/" target="_blank" rel="noopener">Vue 3</a>. What's next? </h3>
</div>"
`;
